jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -381,7 +381,6 @@ object XGBoost extends Serializable {
     val attempt = TaskContext.get().attemptNumber.toString
     rabitEnv.put("DMLC_TASK_ID", taskId)
     rabitEnv.put("DMLC_NUM_ATTEMPT", attempt)
-    rabitEnv.put("DMLC_WORKER_STOP_PROCESS_ON_ERROR", "false")
```

Comment:
well, as I commented in https://github.com/dmlc/rabit/pull/143, this is something we do need to implement current error handling strategy

regarding https://github.com/dmlc/xgboost/issues/4826 , what's your current plan? I think we could replace "kill SparkContext" with "kill the corresponding job" in the case of a task failure in JVM layer

Reply:
@CodingCat It's still on early planing stage and I'm obviously not the right person to talk about spark, so my thought will be largely based on experience with dask.  For spark specific things I usually just forward to @wbo4958 ;-).  I have a few plans that I haven't decided on which one to go yet.  They can be roughly listed as follow:

- We drop the support on single node recovery, and relies on Spark to recover from training failures.  My goal is to have XGBoost fail gracefully without interrupting others.  On dask [a Python exception should do it.](https://distributed.dask.org/en/latest/resilience.html#user-code-failures)  If we do this I want to replace rabit with more matured solutions like MPI.

> "kill the corresponding job" in the case of a task failure in JVM layer

Sounds good to me.  Whatever that doesn't interrupt the others.

- We continue the support on single node recovery, but instead we uses call sequence number as cache key.  My major concern on this is multi-threaded application.  It's not common for users to create DMatrix on their own with the dask interface, but I haven't seen people try to access DMatrix by themselves to bypass dask data loading procedure to avoid extra memory overhead and data balancing.  If they do it  asynchronously then we need serious tests on rabit's thread safety guarantee (which doesn't exist).

Reply:
Either way, the `exit(1)` call has to go, as we need a proper exception.